File: CS-Examples/02_Data/SetArrayOfValuesIntoRange.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;

namespace SetArrayOfValuesIntoRange
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            //Create a workbook.
			Workbook workbook = new Workbook();

            //Create an empty worksheet.
            workbook.CreateEmptySheets(1);

            //Get the worksheet.
            Worksheet sheet = workbook.Worksheets[0];

            //Set the value of max row and column.
            int maxRow = 10000;
            //int maxRow = 5;
            int maxCol = 200;
            //int maxCol = 5;

            //Output an array of data to a range of worksheet.
            object[,] myarray = new object[maxRow + 1, maxCol + 1];
            bool[,] isred = new bool[maxRow + 1, maxCol + 1];
            for (int i = 0; i <= maxRow; i++)
                for (int j = 0; j <= maxCol; j++)
                {
                    myarray[i, j] = i + j;
                    if ((int)myarray[i, j] > 8)
                        isred[i, j] = true;
                }

            // Insert the array of data into the worksheet starting from cell (1, 1)
            sheet.InsertArray(myarray, 1, 1);

            // Specify the name for the resulting Excel file
            String result = "Result-SetArrayOfValuesIntoRange.xlsx";

            // Save the workbook to the specified file in Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the MS Excel file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: CS-Examples/02_Data/SetArrayOfValuesIntoRange.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet

Namespace SetArrayOfValuesIntoRange
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Creates a new Excel workbook.
            Dim workbook As New Workbook()

            ' Creates a single empty worksheet in the workbook.
            workbook.CreateEmptySheets(1)

            ' Retrieves the first worksheet in the workbook (index starts at 0).
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Specifies the maximum number of rows.
            Dim maxRow As Integer = 10000
            ' Specifies the maximum number of columns.
            Dim maxCol As Integer = 200

            ' Declares a 2D array to store data.
            Dim myarray(maxRow, maxCol) As Object
            ' Declares a 2D array to indicate if a cell should be red.
            Dim isred(maxRow, maxCol) As Boolean
            ' Iterates through each row.
            For i As Integer = 0 To maxRow
                ' Iterates through each column.
                For j As Integer = 0 To maxCol
                    ' Computes the value to be assigned to the current cell based on its row and column index.
                    myarray(i, j) = i + j
                    ' Checks if the computed value is greater than 8.
                    If CInt(Fix(myarray(i, j))) > 8 Then
                        ' Sets the corresponding flag to indicate that the cell should be red.
                        isred(i, j) = True
                    End If
                Next j
            Next i
            ' Inserts the array of data into the worksheet starting from cell A1.
            sheet.InsertArray(myarray, 1, 1)
            ' Defines the file name for the resulting Excel file.
            Dim result As String = "Result-SetArrayOfValuesIntoRange.xlsx"

            ' Saves the workbook to a file with the specified file name in the Excel 2013 format.
            workbook.SaveToFile(result, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
